api/query_engine.go

```diff
@@ -3,13 +3,22 @@ package api
 import (
 	"fmt"
 	"sort"
+	"time"
 
 	"github.com/raintank/metrictank/api/models"
 	"github.com/raintank/metrictank/mdata"
+	"github.com/raintank/metrictank/stats"
 	"github.com/raintank/metrictank/util"
 	"github.com/raintank/worldping-api/pkg/log"
 )
 
+var (
+	dpAdjGranularity  = stats.NewGauge32("api.dp_adj_granularity")
+	dpMaxGranularity  = stats.NewGauge32("api.dp_max_granularity")
+	aggAdjGranularity = stats.NewGauge32("api.agg_adj_granularity")
+	aggMaxGranularity = stats.NewGauge32("api.agg_max_granularity")
+)
```

Comment:
these metrics also don't take into account how many series are being requested. alignReq might be called for 1 series, or for >1000, so we should take that into account.